File: models/resource.go
package models

import "time"

type Resource struct {
	ResourceID          string    `json:"resource_id"`
	ResourceName        string    `json:"resource_name"`
	ResourceType        string    `json:"resource_type"`
	RequiredPermissions []string  `json:"required_permissions"`
	Owner               string    `json:"owner"`
	CreatedAt           time.Time `json:"created_at"`
}

File: core-processing/src/main/java/com/taskflow/core/resource/domain/Resource.java
package com.taskflow.core.resource.domain;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Map;

/**
 * 资源 - 领域模型
 */
@Data
@Builder
public class Resource {
    private String resourceId;
    private String tenantId;
    private String type;
    private String name;
    private String description;
    private String status;
    private Map<String, Object> attributes;
    private Map<String, String> labels;
    private String configId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
